module4/projects/solo_stretch_tech/index.md

```diff
@@ -3,14 +3,137 @@ title: Solo Stretch Tech
+<!-- **I used chatGPT for the following list items, please change as needed** -->
+- **GraphQL:** Use GraphQL for querying and mutating data. Students should be able to explain the differences between GraphQL and REST. (This would lead to some changes to how you make requests from the frontend, too!)
+- **Pagination:** Implement pagination to limit the amount of data returned in API responses. Students can use strategies like cursor-based pagination or offset-based pagination. They should discuss why pagination is important for performance and scalability.
+- **[Security Vulnerability Scanner](https://github.com/presidentbeef/brakeman):** Use Brakeman, a static analysis tool, to scan the Rails application for potential security vulnerabilities. Students should be able to explain what Brakeman does, why it’s useful, and how it integrates into the development workflow. 
+- **OpenAPI Documentation:** Use OpenAPI Specification to create comprehensive, interactive API documentation. Students should document all major API endpoints (requests, responses, and parameters) and integrate the documentation using tools like Swagger UI or Redoc.
+- **[Circuit Breaker Design Pattern](https://makisushi.io/posts/asd):** Implement the Circuit Breaker design pattern to handle potential failures when calling external APIs or services. Students should use a gem like stoplight or breaker to demonstrate how the circuit breaker prevents cascading failures and improves app resilience. 
+- **Observability:** Integrate an observability tool (e.g., New Relic, Skylight, Honeybadger) to monitor and analyze the application’s performance, errors, and resource usage. Students should configure the tool to track response times, database queries, and bottlenecks.
+</section>
+
+## Timeline + Deliverables
+
+At the end of the days listed below, you will need to turn in certain deliverables. Your cohort calendar specifies the day of the stretch project so you can easily align when each set of deliverables are due. If you fall behind, talk to your instructor as soon as possible. We want to mimic what communication would look like on the job if you needed more time to complete your assigned tasks.
+
+<section class="note">
+### All deliverables will be turned in via **[this google sheet](https://docs.google.com/spreadsheets/d/1dkhqQctc_6nylYyQII22zLU0He_P1k_xfaJOFngV7dM/edit?usp=sharing)**.
+</section>
+
+<section class="dropdown">
+### Day 2
+- Project Proposal: Make a copy of [this document](https://docs.google.com/document/d/1xUWFfFzSJ09GlWi-55942H2RgDJ8GZN-dTMhNsgakig/edit?usp=sharing) and fill it out. Add the link to your proposal to the google sheet. An instructor will review your proposal. You should not move forward until your proposal has been approved. 
+- Stretch Techs: List the stretch techs you have chosen for your project. 
+</section>
+
+<section class="dropdown">
+### Day 4
```

Comment:
Propose changing this to Day 5. The first week of Stretch coincides with Full Interview simulations. I'm not going to have time to prep interview docs and review project proposals on the same day which would have to happen if we don't want them to move on to these Day 4 (or 5) deliverables before getting instructor approval.

Reply:
I think the rest of the schedule can remain unchanged. They'll still have the whole weekend to work on getting repos set up and issues written for the Day 7 deliverable.